server/src/test/java/org/elasticsearch/search/SearchServiceTests.java

```diff
@@ -1946,6 +1944,63 @@ public void testEnableSearchWorkerThreads() throws IOException {
         }
     }
 
+    public void testEnableQueryPhaseParallelCollection() throws IOException {
+        IndexService indexService = createIndex("index", Settings.EMPTY);
+        IndexShard indexShard = indexService.getShard(0);
+        ShardSearchRequest request = new ShardSearchRequest(
+            OriginalIndices.NONE,
+            new SearchRequest().allowPartialSearchResults(randomBoolean()),
+            indexShard.shardId(),
+            0,
+            indexService.numberOfShards(),
+            AliasFilter.EMPTY,
+            1f,
+            System.currentTimeMillis(),
+            null
+        );
+
+        ThreadPoolExecutor mockThreadPoolExecutor = mock();
+        int executorPoolSize = randomIntBetween(1, 100);
+        when(mockThreadPoolExecutor.getMaximumPoolSize()).thenReturn(executorPoolSize);
+
+        ExecutorService mockNotTPExecutor = mock();
+
+        SearchService service = getInstanceFromNode(SearchService.class);
+        {
+            assertEquals(executorPoolSize, service.determineMaximumNumberOfSlices(mockThreadPoolExecutor, request, ResultsType.DFS));
+            assertEquals(1, service.determineMaximumNumberOfSlices(mockThreadPoolExecutor, request, ResultsType.QUERY));
+            assertEquals(1, service.determineMaximumNumberOfSlices(mockNotTPExecutor, request, ResultsType.DFS));
```

Comment:
What is the value of the additional DFS call with an executor that is not a thread pool executor? Maybe that makes more sense to add to the existing testSupportsParallelCollection. I am thinking you do want to check that the setting you are adding does not affect DFS , which is important and you achieve with the first line.

Reply:
I think I got confused.